src/zkbob/ZkBobPool.sol

```diff
@@ -242,6 +307,143 @@ contract ZkBobPool is EIP1967Admin, Ownable, Parameters, ZkBobAccounting {
         }
     }
 
+    function appendDirectDeposits(
+        uint256 _root_after,
+        uint256[] calldata _indices,
+        uint256 _out_commit,
+        uint256[8] memory _batch_deposit_proof,
+        uint256[8] memory _tree_proof
+    )
+        external
+        onlyOperator
+    {
+        require(_indices.length > 0, "ZkBobPool: empty deposit list");
+        require(_indices.length < 17, "ZkBobPool: too many deposits");
+
+        uint256[33] memory batch_deposit_pub;
+        uint256 total = 0;
+        for (uint256 i = 0; i < _indices.length; i++) {
```

Comment:
So, the DD service is able to skip some deposits and this could be considered as censoring, right?

Reply:
Yes.

I don't see much difference between situations with relayer refusing to process all deposits entirely, or just skipping single user deposits. In both cases, user can request a refund, so there is no difference for them. Yet I think this approach is more flexible from our relayer implementation perspective.